docs/how-to/build-policies/abac/temporary_role.mdx

```diff
@@ -0,0 +1,145 @@
+---
+sidebar_position: 6
+title: Temporary Role Example
+---
+
+Below you'll find how to implement temporary role assignments using ABAC Condition Sets in Permit.io, with a step-by-step implementation guide.
+
+## Prerequisites
+
+Before implementing temporary role assignments, ensure you have:
+- A Permit.io account
+- A tenant attribute called "key" already created and configured
+- A role called "admin" already created 
+- A resource called 'internal_api' already created
+- The admin role assigned to a user in the default and coke tenants
+
+## Implementation Steps
+
+There are several key steps to implement temporary role assignments in Permit.io - all are necessary and should be completed in order.
+
+- ### Add the Tenant attribute "key"
+
+  The first step involves setting up a tenant attribute called "key" and configuring its value.
+  You'll need to access your Permit.io dashboard and navigate through the [Tenant Attributes](https://app.permit.io/user-management/tenant-settings/tenant-attributes).
+  1. [Create Tenant Attributes](https://app.permit.io/user-management/tenant-settings/tenant-attributes)
+      <img src="/img/abac/tenant_key.png" style={{ maxHeight: "45vh" }} />
+  1. [Add key to specific tenant](https://app.permit.io/user-management/tenant-settings/manage-tenants)
+    <img src="/img/abac/add_tenant_key.png" style={{ maxHeight: "45vh" }} />
+  
+  Example for Coke Tenant:
+  ```javascript
+  // Example tenant key attribute value
+  {"key": "coke"}
+  ```
+
+- ### Configure time-based groups
+
+  1. Add the time_based_groups attribute to your user object through the [User Attributes](https://app.permit.io/user-management/tenant-settings/user-attributes) section.
+    <img src="/img/abac/time_based_groups.png" style={{ maxHeight: "45vh" }} />
+  1. Add `time_based_groups` to specific users.
+    <img src="/img/abac/time_based_groups_user.png" style={{ maxHeight: "45vh" }} />
+  1. Make sure the user is [assigned the Top Level Role in the Directory](https://app.permit.io/user-management) as well:
+    <img src="/img/abac/time_based_user_assignment.png" style={{ maxHeight: "45vh" }} />
+
+
+  ```javascript
+  "time_based_groups": [
+    {
+      "role": "admin",
+      "tenant": "default",
+      "expires": 1733256879832
+    },
+    {
+      "role": "admin",
+      "tenant": "coke",
+      "expires": 1733861761081
+    }
+  ]
+  ```
+:::info
+The `expires` value above is the Unix timestamp in milliseconds. This requires that the `current_time` value is passed in the `permit.check()` as a user attribute.
```

Comment:
Worth to add that it's passed in the check method to fit the condition set that you explain in the next step